### PR TITLE
Add background-handler listener, dynamo logging table, change error reporting on failure to add to SQS queue

### PR DESCRIPTION
**Pull request by zach-herridge**
Just an initial idea but adding something like this would let us hook into the important execution events to create the backrgound process history table

### CODE REVIEW

**Comment by zach-herridge** [comment]
So the new implementation is if you want an audit table, create an instance of `BackgroundDynamoAuditTable ` and pass it in with any other listeners you want. Pass the API name or processor instance name into `backgroundQueueName` so you can use the same table for all of the Apis/processors if you want.

**Comment by zach-herridge** [comment]
I changed how the errors bubble up for the SQS queuing. This changes this from a pretty safe PR to a potentially breaking PR if I did something wrong so a closer look might be required and we should probably send this to dev on the other side.

File: src/background-manager.ts [hunk 1]
```diff
@@ -74,34 +76,35 @@ export class BackgroundManager {
...
+        Logger.error('Error inserting background entry into SQS queue : %j', error);
+        throw new Error('Error inserting background entry into SQS queue : ' + error.code + ' : ' + error.name);
+      }
 
-        Logger.info('Background process %s using message id %s', rval, result.MessageId);
+      if (fireStartMessage) {
+        const fireResult: string = await this.fireStartProcessingRequest();
+        Logger.silly('FireResult : %s', fireResult);
       }
-    } catch (err) {
```
Comment (zach-herridge) [inline src/background-manager.ts:103]: I don't think we should be catching errors here. If it doesn't make it into the queue sending the GUI back anyways is confusing.

**Review by bitblit — APPROVED** [review]
LGTM - This is a nice extension - about the only change I'm gonna make is to wrap the listener fires in a try/catch so that if there is a poorly written first one it doesn't halt on it.

**Pull request merged**